coprocessor/fhevm-engine/scheduler/src/dfg/scheduler.rs

```diff
@@ -764,58 +385,270 @@ fn partition_components(
     Ok(())
 }
 
-type TaskResult = Result<(SupportedFheCiphertexts, Option<(i16, Vec<u8>)>)>;
+fn re_randomise_transaction_inputs(
+    inputs: &mut HashMap<Handle, Option<DFGTxInput>>,
+    transaction_id: &Handle,
+    gpu_idx: usize,
+    cpk: tfhe::CompactPublicKey,
+) -> Result<()> {
+    let transaction_rerandomisation_domain_separator = *b"TFHE-w.r";
+    let compact_public_encryption_domain_separator = *b"TFHE.Enc";
+    let mut re_rand_context = ReRandomizationContext::new(
+        transaction_rerandomisation_domain_separator,
+        // First is the function description, second is a nonce
+        [transaction_id.as_slice()],
+        compact_public_encryption_domain_separator,
+    );
+    for txinput in inputs.values_mut() {
+        match txinput {
+            Some(DFGTxInput::Value(val)) => {
+                val.add_to_re_randomization_context(&mut re_rand_context);
+            }
+            Some(DFGTxInput::Compressed((t, c))) => {
+                let decomp = SupportedFheCiphertexts::decompress(*t, c, gpu_idx);
+                if let Ok(decomp) = decomp {
+                    decomp.add_to_rerandomisation_context(&mut re_rand_context);
+                    *txinput = Some(DFGTxInput::Value(decomp));
+                } else {
+                    // If any input fails, we set the input as empty
+                    *txinput = None;
```

Comment:
shouldn't there be an abort or something here ? or return an error ?

it can change the rerand context if it does not fail deterministically across coprocessors I guess

also failing to fetch an input sounds like it should stop the processing of a transaction